### ISSUE TITLE
promisified fs.read() has incorrect return value

### ISSUE THREAD

**Issue by wellcaffeinated** [oven-sh/bun#6863]
### What version of Bun is running?

1.0.7+b0393fba6200d8573f3433fb0af258a0e33ac157

### What platform is your computer?

Darwin 22.6.0 arm64 arm

### What steps can reproduce the bug?

The following test demonstrates the problem.

```js
import fs from 'fs'
import { promisify } from 'util'

const fsread = promisify(fs.read)

async function test(){
  const fd = await fs.promises.open('./index.mjs', 'r')
  const bytes = new Uint8Array(16)
  const ret = await fsread(fd, bytes, 0, 16, 0)
  console.log('ret is', ret)
  console.log('bytes read', ret.bytesRead)
}

test()
```

Output is:
> ret is 16
> bytes read undefined

### What is the expected behavior?

According to the [node.js docs](https://nodejs.org/api/fs.html#fsreadfd-buffer-offset-length-position-callback): 

> If this method is invoked as its [util.promisify()](https://nodejs.org/api/util.html#utilpromisifyoriginal)ed version, it returns a promise for an Object with bytesRead and buffer properties.

In other words, it should return `{ bytesRead: 16, buffer }`

### What do you see instead?

Instead it returns integer of bytes read

**Comment by AlttiRi** [comment]
The same is with [`fs.promises.open`](https://nodejs.org/api/fs.html#fspromisesopenpath-flags-mode).

It should return [`FileHandle`](https://nodejs.org/api/fs.html#class-filehandle). But, based on your code, it returns `fd` (`number`).

In Node.js your code does not work due to passing `FileHandle` instead of `number` to [`fs.read`](https://nodejs.org/api/fs.html#fsreadfd-buffer-offset-length-position-callback).
```
TypeError [ERR_INVALID_ARG_TYPE]: The "fd" argument must be of type number. Received an instance of FileHandle
```

Here is the correct one:
```js
async function test() {
    const fileHandle = await fs.promises.open("./index.mjs", "r");
    const bytes = new Uint8Array(16);

    const result = await fsread(fileHandle.fd, bytes, 0, 16, 0);
    console.log(result);
    console.log(result.bytesRead); // 16
    console.log(result.buffer);    // Uint8Array(16) [ ... ]
}
```

---
UPD:  
It's a known issue: #3167 #5918
